### PR TITLE
Feature/port to agl

### PR DESCRIPTION
**Pull request by mkorniichuk**
This PR is **ready** for review.

### Summary
Current PR includes changes which allow to use cross-compilation in order to build SDL for AGL-based platform

##### Breaking Changes
* Third party structural and logical changes
* Toolchain support
* Toolchain for AGL

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/3rd_party/apache-log4cxx/helpers.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    CURRENT_VERSION=$(cd $LOG4CXX_SOURCE_DIRECTORY && git log --pretty=format:%H -1 . 2>/dev/null)
+    if [[ $INSTALLED_VERSION == $CURRENT_VERSION ]]; then
+        echo "TRUE"
+    else
+        echo "FALSE"
+    fi
+}
+
+
+      #   COMMAND /bin/bash -c \"cd ${CMAKE_CURRENT_SOURCE_DIR} &&
```
Comment (LuxoftAKutsan) [inline src/3rd_party/apache-log4cxx/helpers.sh]: @mkorniichuk old code?

File: src/3rd_party/apache-log4cxx/helpers.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+#
```
Comment (LuxoftAKutsan) [inline src/3rd_party/apache-log4cxx/helpers.sh:3]: @mkorniichuk please add to comment main functions and required variables, and some usage examples

File: src/3rd_party/apache-log4cxx/helpers.sh [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+DEBUG() {
+    echo "$@"
+}
+
+AppendCommitHash() {
+    LOG4CXX_SOURCE_DIRECTORY=$1
+    LOG4CXX_LIBS_DIRECTORY=$2
+    COMMIT_HASH_FILE=/tmp/commit_hash
+    cd $LOG4CXX_SOURCE_DIRECTORY && git log --pretty=format:%H -1 . 1>$COMMIT_HASH_FILE 2>/dev/null
```
Comment (LuxoftAKutsan) [inline src/3rd_party/apache-log4cxx/helpers.sh]: @mkorniichuk do we rely need to mute errors?

File: tools/infrastructure/docker/AGL/workspace/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,82 @@
+FROM ubuntu:16.04
```
Comment (LuxoftAKutsan) [inline tools/infrastructure/docker/AGL/workspace/Dockerfile]: @mkorniichuk I would avoid committing workplace for AGL in repo

**Review by LuxoftAKutsan — APPROVED** [review]
:+1: 
Please format all your **new** changes with cmake-format

File: tools/infrastructure/docker/AGL/workspace/Dockerfile (hunk 4 above)
Comment (mkorniichuk) [reply tools/infrastructure/docker/AGL/workspace/Dockerfile]: @LuxoftAKutsan fixed within 472c3a9a75a1670991f453d0e6453cb371e03e3b

File: src/3rd_party/apache-log4cxx/helpers.sh (hunk 1 above)
Comment (mkorniichuk) [reply src/3rd_party/apache-log4cxx/helpers.sh]: @LuxoftAKutsan fixed within bf0ee058043a3c5988f9ca2f28ecf5c31183842f

File: src/3rd_party/apache-log4cxx/helpers.sh (hunk 2 above)
Comment (mkorniichuk) [reply src/3rd_party/apache-log4cxx/helpers.sh:3]: @LuxoftAKutsan fixed within bf0ee058043a3c5988f9ca2f28ecf5c31183842f

File: src/3rd_party/apache-log4cxx/helpers.sh (hunk 3 above)
Comment (mkorniichuk) [reply src/3rd_party/apache-log4cxx/helpers.sh]: @LuxoftAKutsan fixed within bf0ee058043a3c5988f9ca2f28ecf5c31183842f

File: CMakeLists.txt [hunk 5]
```diff
@@ -290,9 +282,8 @@ if(ENABLE_GCOV)
   message(STATUS "GCOV enabled")
 endif()
 
-set(CMAKE_CXX_FLAGS_RELEASE " -DNDEBUG -s -O2")
-set(CMAKE_CXX_FLAGS_DEBUG   " -DDEBUG -g3 -ggdb3")
-
+# TODO(AK): check current OS here
+add_definitions(-DOS_POSIX) # see line :360
```
Comment (LuxoftAKutsan) [inline CMakeLists.txt]: @mkorniichuk line 360 : 
```
if (HMI STREQUAL "web" AND NOT DEFINED HMIADAPTER)
```
Please avoid line numbers in comments

File: agl_linux_x86_64.cmake [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+set(TARGET_PLATFORM AGL)
+
+include(/opt/agl-sdk/6.0.2-corei7-64/sysroots/x86_64-aglsdk-linux/usr/share/cmake/OEToolchainConfig.cmake)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
```
Comment (LuxoftAKutsan) [inline agl_linux_x86_64.cmake]: @mkorniichuk position independent code may be setted up with cmake
DOne in #11

File: CMakeLists.txt [hunk 7]
```diff
@@ -224,27 +215,28 @@ add_custom_target(genivi-tarball
...
 
 if (USE_GOLD_LD)
-  execute_process(COMMAND ld -v OUTPUT_VARIABLE result)
-  string(FIND ${result} Binutils GOLD_LD_FOUND)
-  if (GOLD_LD_FOUND EQUAL -1)
-    message(SEND_ERROR "Gold linker not found!")
-  else()
+  execute_process(COMMAND ${CMAKE_CXX_COMPILER} -fuse-ld=gold -Wl,--version ERROR_QUIET OUTPUT_VARIABLE GOLD_VERSION)
+  if(GOLD_VERSION MATCHES "GNU gold")
     message(INFO " Use gold linker")
```
Comment (LuxoftAKutsan) [inline CMakeLists.txt]: @mkorniichuk replace INFO to status. There is no such message threshold as INFO

File: tools/infrastructure/docker/AGL/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,48 @@
+FROM ubuntu:16.04
+
+RUN apt-get update && apt-get install -yq \
+    sudo \
+    git \
+    wget \
+    vim \
```
Comment (LuxoftAKutsan) [inline tools/infrastructure/docker/AGL/Dockerfile]: @mkorniichuk remove redundant tools from docker file that should only build SDL

**Review by LuxoftAKutsan — APPROVED** [review]
:+1:  Please fix minor issues

File: tools/infrastructure/docker/AGL/Dockerfile (hunk 8 above)
Comment (mkorniichuk) [reply tools/infrastructure/docker/AGL/Dockerfile]: @LuxoftAKutsan  fixed within 0e2904f7b83ae3e408736ef42ddbd3d8d833bb68

File: CMakeLists.txt (hunk 7 above)
Comment (mkorniichuk) [reply CMakeLists.txt]: @LuxoftAKutsan fixed within a89e4042f9a29e6b3ec306046c57bd2bd2e2c4f7

File: agl_linux_x86_64.cmake (hunk 6 above)
Comment (mkorniichuk) [reply agl_linux_x86_64.cmake]: @LuxoftAKutsan fixed within a89e4042f9a29e6b3ec306046c57bd2bd2e2c4f7

File: CMakeLists.txt (hunk 5 above)
Comment (mkorniichuk) [reply CMakeLists.txt]: @LuxoftAKutsan fixed within a89e4042f9a29e6b3ec306046c57bd2bd2e2c4f7

File: src/3rd_party/apache-log4cxx/AGL.cmake [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  "--with-apr-util=${APR_UTIL_BUILD_DIR}"
+  "LDFLAGS=-L${EXPAT_LIBS_DIRECTORY}"
+)
+
+add_custom_command(OUTPUT ${LOG4CXX_BUILD_DIRECTORY}/Makefile
+  COMMAND ${LOG4CXX_SOURCE_DIRECTORY}/configure ${CONFIGURE_FLAGS}
+  DEPENDS libapr-1
+  DEPENDS apr-util
+  WORKING_DIRECTORY ${LOG4CXX_BUILD_DIRECTORY}
+)
```
Comment (BSolonenko) [inline src/3rd_party/apache-log4cxx/AGL.cmake]: @mkorniichuk please add empty line at the end of the file

File: src/3rd_party/apache-log4cxx/AGL.cmake [hunk 10]
```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2018, Ford Motor Company
```
Comment (BSolonenko) [inline src/3rd_party/apache-log4cxx/AGL.cmake]: @mkorniichuk Copyright (c) **2019**, Ford Motor Company

**Comment by vkushnirenko-luxoft** [comment]
Linux_build

**Review by IGapchuk — APPROVED** [review]

File: src/3rd_party/boost/Linux.cmake [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+if (${3RD_PARTY_INSTALL_PREFIX} MATCHES "/usr/local")
+  set(INSTALL_COMMAND "sudo ${INSTALL_COMMAND}")
+endif()
+
+install(
+  CODE "execute_process(
+    WORKING_DIRECTORY ${BOOST_SOURCE_DIRECTORY} 
+    COMMAND /bin/bash -c \"${INSTALL_COMMAND}\"
+  )"
+)
```
Comment (SNiukalov) [inline src/3rd_party/boost/Linux.cmake]: Please add empty line at end file

File: src/3rd_party/apr/QNX.cmake [hunk 12]
```diff
@@ -51,3 +51,16 @@ set(CONFIGURE_FLAGS
...
+  COMMAND ${CMAKE_COMMAND} -E copy include/apr.h ${APR_SOURCE_DIRECTORY}/include
+  WORKING_DIRECTORY ${APR_BUILD_DIRECTORY}
+)
+
+add_custom_command(OUTPUT ${APR_BUILD_DIRECTORY}/include/private/apr_escape_test_char.h
+  DEPENDS ${APR_BUILD_DIRECTORY}/Makefile
+  COMMAND make include/private/apr_escape_test_char.h
+  COMMAND ${CMAKE_COMMAND} -E copy include/private/apr_escape_test_char.h ${APR_SOURCE_DIRECTORY}/include/private
+  WORKING_DIRECTORY ${APR_BUILD_DIRECTORY}
+)
```
Comment (SNiukalov) [inline src/3rd_party/apr/QNX.cmake]: Please add empty line at end file

**Review by Ypostolov — APPROVED** [review]

**Review by Ypostolov — APPROVED** [review]

**Review by Ypostolov — APPROVED** [review]

**Review by Ypostolov — APPROVED** [review]

**Review by Ypostolov — APPROVED** [review]

File: CMakeLists.txt [hunk 13]
```diff
@@ -304,26 +290,22 @@ if(ENABLE_GCOV)
   message(STATUS "GCOV enabled")
 endif()
 
-set(CMAKE_CXX_FLAGS_RELEASE " -DNDEBUG -s -O2")
-set(CMAKE_CXX_FLAGS_DEBUG   " -DDEBUG -g3 -ggdb3")
-
+# TODO(AK): check current OS here
```
Comment (AByzhynar) [inline CMakeLists.txt]: @mkorniichuk What do we need `TODO` here for?

File: src/3rd_party/apache-log4cxx/Linux.cmake [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+  "--with-apr-util=${APR_UTIL_BUILD_DIR}"
+  "LDFLAGS=-L${EXPAT_LIBS_DIRECTORY}"
+)
+
+add_custom_command(OUTPUT ${LOG4CXX_BUILD_DIRECTORY}/Makefile
+  COMMAND CC=${CMAKE_C_COMPILER} CXX=${CMAKE_CXX_COMPILER} ${LOG4CXX_SOURCE_DIRECTORY}/configure ${CONFIGURE_FLAGS}
+  DEPENDS libapr-1
+  DEPENDS apr-util
+  WORKING_DIRECTORY ${LOG4CXX_BUILD_DIRECTORY}
+)
```
Comment (AByzhynar) [inline src/3rd_party/apache-log4cxx/Linux.cmake]: @mkorniichuk 
Please add an empty line here

File: src/3rd_party/apache-log4cxx/QNX7.cmake [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+  "CXX=${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-g++${HOST_EXECUTABLE_SUFFIX}"
+  "CC=${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-gcc${HOST_EXECUTABLE_SUFFIX}"
+)
+
+add_custom_command(OUTPUT ${LOG4CXX_BUILD_DIRECTORY}/Makefile
+  COMMAND ${LOG4CXX_SOURCE_DIRECTORY}/configure ${CONFIGURE_FLAGS}
+  DEPENDS libapr-1
+  DEPENDS apr-util
+  WORKING_DIRECTORY ${LOG4CXX_BUILD_DIRECTORY}
+)
```
Comment (AByzhynar) [inline src/3rd_party/apache-log4cxx/QNX7.cmake]: @mkorniichuk Check all files for empty line presence at the end

File: src/3rd_party/apache-log4cxx/helpers.sh (hunk 2 above)
Comment (AByzhynar) [reply src/3rd_party/apache-log4cxx/helpers.sh:3]: @mkorniichuk I would append with helper command line options `-h`, `--help` to get help

File: src/3rd_party/apr-util/Linux.cmake [hunk 16]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2014, Ford Motor Company
+# Copyright (c) 2018, Ford Motor Company
```
Comment (AByzhynar) [inline src/3rd_party/apr-util/Linux.cmake]: @mkorniichuk If you decided to change a year - change it to 2019 and in all touched files

File: src/3rd_party/apache-log4cxx/helpers.sh [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+IsActualVersion() {
+    LOG4CXX_SOURCE_DIRECTORY=$1
+    LOG4CXX_LIBS_DIRECTORY=$2
+    INSTALLED_VERSION=$(readelf -p .commit_hash ${LOG4CXX_LIBS_DIRECTORY%/}/liblog4cxx.so | awk 'NR==3 {print $NF}')
+    CURRENT_VERSION=$(cd $LOG4CXX_SOURCE_DIRECTORY && git log --pretty=format:%H -1 .)
+    if [[ $INSTALLED_VERSION == $CURRENT_VERSION ]]; then
+        return 1
+    else
```
Comment (AByzhynar) [inline src/3rd_party/apache-log4cxx/helpers.sh]: @mkorniichuk redundant `else`

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/3rd_party/apache-log4cxx/CMakeLists.txt [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR})
+
+set(LOG4CXX_SOURCE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/apache-log4cxx-0.10.0)
+set(LOG4CXX_USE_SYSTEM_PATH 0)
+find_package(Log4cxx)
+
+if ((LOG4CXX_FOUND OR NO_REBUILD_3RD_PARTY_LOGGER) AND
+    NOT FORCE_3RD_PARTY_LOGGER)
+
```
Comment (ZhdanovP) [inline src/3rd_party/apache-log4cxx/CMakeLists.txt]: @mkorniichuk Extra line?

File: src/3rd_party/apr-util/AGL.cmake [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+  "--with-expat-source=${EXPAT_SOURCE_DIRECTORY}"
+  "--with-expat-build=${EXPAT_BUILD_DIRECTORY}"
+)
+
+add_custom_command(OUTPUT ${APR_UTIL_BUILD_DIRECTORY}/Makefile
+  COMMAND ${APR_UTIL_SOURCE_DIRECTORY}/configure ${CONFIGURE_FLAGS}
+  DEPENDS libapr-1
+  DEPENDS expat
+  WORKING_DIRECTORY ${APR_UTIL_BUILD_DIRECTORY}
+)
```
Comment (ZhdanovP) [inline src/3rd_party/apr-util/AGL.cmake]: @mkorniichuk Need line in the file end

File: src/3rd_party/apr-util/CMakeLists.txt [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+    WORKING_DIRECTORY ${APR_UTIL_BUILD_DIRECTORY}
+  )
+
+  install(
+    DIRECTORY ${APR_UTIL_LIBS_DIRECTORY}/
+    DESTINATION ${3RD_PARTY_INSTALL_PREFIX_ARCH}/lib
+    USE_SOURCE_PERMISSIONS
+    FILES_MATCHING PATTERN libaprutil-1.so*
+  )
+endif()
```
Comment (ZhdanovP) [inline src/3rd_party/apr-util/CMakeLists.txt]: @mkorniichuk add line

File: src/3rd_party/apr-util/Linux.cmake [hunk 21]
```diff
@@ -28,46 +28,15 @@
...
-  WORKING_DIRECTORY ${APR_UTIL_BUILD_DIRECTORY}
-)
-
-install(
-  DIRECTORY ${APR_UTIL_LIBS_DIRECTORY}/
-  DESTINATION ${3RD_PARTY_INSTALL_PREFIX_ARCH}/lib
-  USE_SOURCE_PERMISSIONS
-  FILES_MATCHING PATTERN libaprutil-1.so*
-)
+)
```
Comment (ZhdanovP) [inline src/3rd_party/apr-util/Linux.cmake]: @mkorniichuk add line

File: src/3rd_party/apr/CMakeLists.txt [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+# POSSIBILITY OF SUCH DAMAGE.
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR})
+
+set(APR_FIND_VERSION 1.5.0)
+set(APR_USE_SYSTEM_PATH 0)
+find_package(APR)
+
+if ((APR_FOUND OR NO_REBUILD_3RD_PARTY_LOGGER) AND
+    NOT FORCE_3RD_PARTY_LOGGER)
```
Comment (ZhdanovP) [inline src/3rd_party/apr/CMakeLists.txt:38]: @mkorniichuk seems like we can have all condition on one line and remove redundant line 39

File: src/3rd_party/apr/Linux.cmake [hunk 23]
```diff
@@ -61,23 +39,4 @@ add_custom_command(OUTPUT ${APR_BUILD_DIRECTORY}/include/private/apr_escape_test
...
-  FILES_MATCHING PATTERN libapr-1.so*
-)
-
-install(
-  DIRECTORY ${APR_INCLUDE_DIRECTORY}/
-  DESTINATION ${3RD_PARTY_INSTALL_PREFIX}/include
-  USE_SOURCE_PERMISSIONS
-  FILES_MATCHING PATTERN *.h
-)
+)
```
Comment (ZhdanovP) [inline src/3rd_party/apr/Linux.cmake]: @mkorniichuk add line

File: src/3rd_party/apr/QNX7.cmake [hunk 24]
```diff
@@ -51,3 +51,16 @@ set(CONFIGURE_FLAGS
...
+  COMMAND ${CMAKE_COMMAND} -E copy include/apr.h ${APR_SOURCE_DIRECTORY}/include
+  WORKING_DIRECTORY ${APR_BUILD_DIRECTORY}
+)
+
+add_custom_command(OUTPUT ${APR_BUILD_DIRECTORY}/include/private/apr_escape_test_char.h
+  DEPENDS ${APR_BUILD_DIRECTORY}/Makefile
+  COMMAND make include/private/apr_escape_test_char.h
+  COMMAND ${CMAKE_COMMAND} -E copy include/private/apr_escape_test_char.h ${APR_SOURCE_DIRECTORY}/include/private
+  WORKING_DIRECTORY ${APR_BUILD_DIRECTORY}
+)
```
Comment (ZhdanovP) [inline src/3rd_party/apr/QNX7.cmake]: @mkorniichuk add line

File: src/3rd_party/boost/AGL.cmake [hunk 25]
```diff
@@ -0,0 +1,66 @@
...
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+include(ExternalProject)
+
+set(BOOST_AGL_PROJECT_CONFIG_JAM
+  "using gcc : agl : x86_64-agl-linux-gcc  -march=corei7 -mtune=corei7 -mfpmath=sse -msse4.2 --sysroot=/opt/agl-sdk/6.0.2-corei7-64/sysroots/corei7-64-agl-linux : ")
+
+set(CONFIGURE_COMMAND
+  CC=\"\" ./bootstrap.sh --with-libraries=system,thread,date_time,filesystem --prefix=${3RD_PARTY_INSTALL_PREFIX} COMMAND echo ${BOOST_AGL_PROJECT_CONFIG_JAM} $<SEMICOLON> >> ./project-config.jam)
```
Comment (ZhdanovP) [inline src/3rd_party/boost/AGL.cmake]: @mkorniichuk Seems like too long line, do we have any constraints of symbols in one line?

File: src/3rd_party/boost/FindBoost.cmake [hunk 26]
```diff
@@ -0,0 +1,283 @@
...
+  math(EXPR Boost_SUBMINOR_VERSION "${_Boost_VERSION} % 100")
+  unset(_Boost_VERSION)
+
+  set(Boost_VERSION "${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION}")
+  if (Boost_VERSION VERSION_LESS BOOST_FIND_VERSION)
+    message(WARNING "Detected version of Boost is too old (${Boost_VERSION}). Requested version was ${BOOST_FIND_VERSION}")
+    set(Boost_FOUND 0)
+  endif()
+
+  ###########################
```
Comment (ZhdanovP) [inline src/3rd_party/boost/FindBoost.cmake]: @mkorniichuk Do we need this fence here?

File: src/3rd_party/boost/QNX7.cmake [hunk 27]
```diff
@@ -28,18 +28,17 @@
...
   sed -Ei "s/case darwin/case *qnx*/g" ./tools/build/src/tools/gcc.jam)
-set(
-  BOOST_FILESYSTEM_OPERATION
+set(BOOST_FILESYSTEM_OPERATION
   sed -Ei "s/__SUNPRO_CC/__QNX__/g" ./libs/filesystem/src/operations.cpp)
-set(
-BOOTSTRAP
-./bootstrap.sh --with-toolset=gcc --with-libraries=system,thread,date_time,filesystem --prefix=${3RD_PARTY_INSTALL_PREFIX})
+set(BOOTSTRAP
+  ./bootstrap.sh --with-toolset=gcc --with-libraries=system,thread,date_time,filesystem --prefix=${3RD_PARTY_INSTALL_PREFIX})
```
Comment (ZhdanovP) [inline src/3rd_party/boost/QNX7.cmake]: @mkorniichuk This file contains a lot of long lines, it's seems to be hard to perception.

File: src/3rd_party/bson/QNX7.cmake [hunk 28]
```diff
@@ -44,4 +46,34 @@
...
+if (${3RD_PARTY_INSTALL_PREFIX} MATCHES "/usr/local")
+  set(INSTALL_COMMAND "sudo ${INSTALL_COMMAND}")
+endif()
+
+install(
+  CODE "execute_process(
+    WORKING_DIRECTORY ${BSON_LIB_SOURCE_DIRECTORY}
+    COMMAND /bin/bash -c \"${INSTALL_COMMAND}\"
+  )"
+)
```
Comment (ZhdanovP) [inline src/3rd_party/bson/QNX7.cmake]: @mkorniichuk Additional line

File: src/appMain/build_config.txt [hunk 29]
```diff
@@ -1,5 +1,5 @@
 //Policy mode (PROPRIETARY, HTTP or EXTERNAL_PROPRIETARY)
-EXTENDED_POLICY:STRING=PROPRIETARY
+EXTENDED_POLICY:STRING=EXTERNAL_PROPRIETARY
```
Comment (ZhdanovP) [inline src/appMain/build_config.txt]: @mkorniichuk Will we have EXTERNAL_PROPRIETARY policy by default? Why do you change it?

File: tools/infrastructure/docker/QNX/entrypoint.sh [hunk 30]
```diff
@@ -0,0 +1,24 @@
...
+
+EX_CODE=$(id -u developer > /dev/null 2&>1 || echo $?)
+if [[ $EX_CODE -ne $? ]]; then
+    useradd --shell /bin/bash -u $USER_ID -o -c "" developer
+    echo "developer ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+    chown -R developer:developer /home/developer
+    export HOME=/home/developer
+fi
+
+echo "Welcome to developer_qnx:build!"
```
Comment (ZhdanovP) [inline tools/infrastructure/docker/QNX/entrypoint.sh]: @alignment alignment if you want

**Review by ZhdanovP — APPROVED** [review]

**Comment by ZhdanovP** [comment]
@mkorniichuk Why the "Linux_build" on CI are failed?

**Comment by vkushnirenko-luxoft** [comment]
Linux_build

File: src/3rd_party/apache-log4cxx/helpers.sh (hunk 2 above)
Comment (mkorniichuk) [reply src/3rd_party/apache-log4cxx/helpers.sh:3]: @AByzhynar cmake will use this script, not a developer, so a simple comment is enough.

File: tools/infrastructure/docker/QNX/entrypoint.sh (hunk 30 above)
Comment (mkorniichuk) [reply tools/infrastructure/docker/QNX/entrypoint.sh]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/appMain/build_config.txt (hunk 29 above)
Comment (mkorniichuk) [reply src/appMain/build_config.txt]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/bson/QNX7.cmake (hunk 28 above)
Comment (mkorniichuk) [reply src/3rd_party/bson/QNX7.cmake]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/boost/QNX7.cmake (hunk 27 above)
Comment (mkorniichuk) [reply src/3rd_party/boost/QNX7.cmake]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/boost/FindBoost.cmake (hunk 26 above)
Comment (mkorniichuk) [reply src/3rd_party/boost/FindBoost.cmake]: @ZhdanovP removed in ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apr/QNX7.cmake (hunk 24 above)
Comment (mkorniichuk) [reply src/3rd_party/apr/QNX7.cmake]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apr/Linux.cmake (hunk 23 above)
Comment (mkorniichuk) [reply src/3rd_party/apr/Linux.cmake]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apr/CMakeLists.txt (hunk 22 above)
Comment (mkorniichuk) [reply src/3rd_party/apr/CMakeLists.txt:38]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apr-util/Linux.cmake (hunk 21 above)
Comment (mkorniichuk) [reply src/3rd_party/apr-util/Linux.cmake]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apr-util/CMakeLists.txt (hunk 20 above)
Comment (mkorniichuk) [reply src/3rd_party/apr-util/CMakeLists.txt]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apr-util/AGL.cmake (hunk 19 above)
Comment (mkorniichuk) [reply src/3rd_party/apr-util/AGL.cmake]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apache-log4cxx/CMakeLists.txt (hunk 18 above)
Comment (mkorniichuk) [reply src/3rd_party/apache-log4cxx/CMakeLists.txt]: @ZhdanovP fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apache-log4cxx/helpers.sh (hunk 17 above)
Comment (mkorniichuk) [reply src/3rd_party/apache-log4cxx/helpers.sh]: @AByzhynar fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apr-util/Linux.cmake (hunk 16 above)
Comment (mkorniichuk) [reply src/3rd_party/apr-util/Linux.cmake]: @AByzhynar fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apache-log4cxx/QNX7.cmake (hunk 15 above)
Comment (mkorniichuk) [reply src/3rd_party/apache-log4cxx/QNX7.cmake]: @AByzhynar fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: src/3rd_party/apache-log4cxx/Linux.cmake (hunk 14 above)
Comment (mkorniichuk) [reply src/3rd_party/apache-log4cxx/Linux.cmake]: @AByzhynar fixed within ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

File: CMakeLists.txt (hunk 13 above)
Comment (mkorniichuk) [reply CMakeLists.txt]: @AByzhynar removed in ae7d52ab0f457d8f46b1ce7aedaf336c3f9d1933

**Comment by vkushnirenko-luxoft** [comment]
Linux_build

File: src/3rd_party/boost/AGL.cmake (hunk 25 above)
Comment (mkorniichuk) [reply src/3rd_party/boost/AGL.cmake]: @ZhdanovP fixed within 160e7a336dbc56913b91e128daabf59f3b69933e

**Comment by vkushnirenko-luxoft** [comment]
Linux_build

**Review by BSolonenko — APPROVED** [review]

**Review by AByzhynar — APPROVED** [review]

**Pull request merged**